util/encoding/encoding.go

```diff
@@ -297,3 +293,49 @@ func DecodeVarUint64Decreasing(b []byte) ([]byte, uint64) {
 	leftover, v := DecodeVarUint64(b)
 	return leftover, ^v
 }
+
+const (
+	escape = 0x00
+	term   = 0x01
+	null   = 0xff
+)
+
+// EncodeBytes ...
```

Comment:
Finish documenting these methods, including a description of the encoding and the way the numeric values were chosen to preserve ordering.